### PR TITLE
[1.13] Modified BlockBubbleColumn so mod-added blocks can create bubble columns

### PR DESCRIPTION
**Pull request by Laton95**
Vanilla BlockBubbleColumn destroys itself when it is not placed on either soul sand or magma blocks. I've added two methods in IForgeBlock and IForgeBlockState to replace those hardcoded blocks, similar to how beacons were modified, so mod-added blocks can create bubble columns on top of them.

Example mod here:
https://pastebin.com/PZ1d7myy

### CODE REVIEW

**Comment by bs2609** [comment]
This seems like something that can potentially be better implemented by defining a tag for it.

**Comment by Laton95** [comment]
> This seems like something that can potentially be better implemented by defining a tag for it.

If tags are the new way to go, sure. I was just working off the precedent of isBeaconBase, but maybe that should change in the future too

**Comment by liach** [comment]
you should add another tag for columns that push, and make the overall tag reference the 2 subtags instead of individual blocks.

**Comment by Laton95** [comment]
> you should add another tag for columns that push, and make the overall tag reference the 2 subtags instead of individual blocks.

Good idea, and done.

I want to make it clear though that these tags will not cause any block to create bubbles by themselves. It would be up to the modder to place the bubble column block, these changes just allow changes to what blocks bubble columns think are valid to be placed on.

**Review by liach — APPROVED** [review]

**Comment by liach** [comment]
looks good, why no merge?

File: patches/minecraft/net/minecraft/block/BlockBubbleColumn.java.patch [hunk 1]
```diff
@@ -0,0 +1,22 @@
+--- a/net/minecraft/block/BlockBubbleColumn.java
++++ b/net/minecraft/block/BlockBubbleColumn.java
+@@ -86,7 +86,7 @@
+       if (block == Blocks.field_203203_C) {
+          return iblockstate.func_177229_b(field_203160_a);
+       } else {
+-         return block != Blocks.field_150425_aM;
++         return p_203157_0_.func_180495_p(p_203157_1_.func_177977_b()).func_203425_a(net.minecraftforge.common.Tags.Blocks.BUBBLE_SOURCES_WITH_DRAG);
```
Comment (mezz) [inline patches/minecraft/net/minecraft/block/BlockBubbleColumn.java.patch:8]: I'm not sure this is equivalent. The original `block` comes from `iblockstate` which is at position `p_203157_1_`.
Why are you checking the block state below `p_203157_1_` here?

**Review by mezz — CHANGES_REQUESTED** [review]

**Pull request closed without merging**